pulsar-broker/src/main/java/org/apache/pulsar/broker/resourcegroup/ResourceGroupPublishLimiter.java

```diff
@@ -160,12 +168,22 @@ private void replaceLimiters(Runnable updater) {
 
     @Override
     public void close() {
-      this.apply();
-      replaceLimiters(null);
+        replaceLimiters(null);
```

Comment:
Does the order of calls matter here? If so maybe add a comment?

Reply:
@cckellogg thank you. I added the comment. I also fixed the case where the callback was not getting called when ratelimiter is closed (not invoking the callback can leave the producer blocked forever). PTAL.